Note: import amended to just required symbols to satify Perl 5.30.
  Subroutine Mirmon::pack_sockaddr_in6 redefined at /usr/lib64/perl5/5.30.1/Exporter.pm line 66.
   at /usr/share/webapps/mirmon/2.10/hostroot/mirmon line 175.

https://bugs.gentoo.org/show_bug.cgi?id=289943

From ab302148275a973c3fccd164fa1ae4b4c4409d21 Mon Sep 17 00:00:00 2001
From: Jeremy Olexa <darkside@gentoo.org>
Date: Wed, 28 Oct 2009 21:04:39 -0500
Subject: [PATCH 2/2] Add ipv6 monitor support to mirmon

This patch will overload the gethost() function provided by perl in such a way
that it uses gethostbyname2() from Socket6, which is a new dependency.

Original patch by: Kent Fredric
---
 mirmon |   12 +++++++++++-
 1 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/mirmon b/mirmon
index 6c0aaa8..50d40ca 100755
--- a/mirmon
+++ b/mirmon
@@ -30,7 +30,17 @@ my $VER = '# $Id$
 use strict ;
 use IO::Pipe ;
 use IO::Select ;
-use Net::hostent ;
+
+use Socket qw(AF_INET);
+use Socket6 qw(AF_INET6);
+use Data::Dumper qw( Dumper );
+
+sub gethost($){
+  my $url = shift;
+  return 1 if gethostbyname2 $url, AF_INET;
+  return 1 if gethostbyname2 $url, AF_INET6;
+  return 0;
+}
 
 my $DEF_CNF = "/etc/$PRG.conf" ;
 
-- 
1.6.4.4

